compiler/src/iree/compiler/DispatchCreation/CollapseDimensions.cpp

```diff
@@ -405,19 +413,114 @@ CollapseInfo::getTransformedCollapsableLoops(AffineMap map) const {
   return transformedLoops;
 }
 
-// Update `collapsableLoops` by taking the set intersection with
-// `otherCollapsable` and update the reassociation indicies accordingly.
-bool CollapseInfo::updateCollapseViaIntersect(
-    const CollapsableLoopsSet &otherCollapsable) {
-  CollapsableLoopsSet toRemove;
-  for (auto elem : collapsableLoops) {
-    if (!otherCollapsable.contains(elem)) {
-      toRemove.insert(elem);
+SmallVector<ReassociationIndices>
+CollapseInfo::getTransformedReassociation(AffineMap map) const {
+  SmallVector<ReassociationIndices> transformedReassociation(
+      reassociation.size());
+  for (const auto &[i, indicies] : llvm::enumerate(reassociation)) {
+    for (auto elem : indicies) {
+      auto dimExpr = dyn_cast<AffineDimExpr>(map.getResult(elem));
+      if (!dimExpr) {
+        break;
+      }
+      transformedReassociation[i].push_back(dimExpr.getPosition());
     }
   }
-  collapsableLoops.set_subtract(toRemove);
-  updateReassociation();
-  return toRemove.size();
+  return transformedReassociation;
+}
+
+bool CollapseInfo::updateFromConsumer(OpOperand *operand,
+                                      const CollapseInfo &consumerInfo) {
+  FailureOr<AffineMap> consumerToProducerMap =
+      getConsumerLoopToProducerLoopsMap(*operand);
+  if (failed(consumerToProducerMap)) {
+    return this->clear();
+  }
+
+  FailureOr<CollapsableLoopsSet> consumerCollapsable =
+      consumerInfo.getTransformedCollapsableLoops(
+          consumerToProducerMap.value());
+  if (failed(consumerCollapsable)) {
+    return clear();
+  }
+
+  FailureOr<SmallVector<ReassociationIndices>> consumerReassoc =
+      consumerInfo.getTransformedReassociation(consumerToProducerMap.value());
+  assert(succeeded(consumerReassoc));
```

Comment:
I think we should remove the `FailureOr`?

```suggestion
  SmallVector<ReassociationIndices> consumerReassoc =
      consumerInfo.getTransformedReassociation(consumerToProducerMap.value());
```

Reply:
I was confused that why it compiles, then I understood. The `getTransformedReassociation` returns `SmallVector<ReassociationIndices>`, so it will always succeeded. lol

Reply:
Yes, thank you. Forgot to remove this